HOMEWORK_2

gavrilenco opened the pull request:
- added form to CommentList
	- added tests to Article and CommentList
	- added propTypes

Commit:
HOMEWORK_2

	- added form to CommentList
	- added tests to Article and CommentList
	- added propTypes

romabelka commented on src/components/article.spec.js:
```diff
@@ -0,0 +1,69 @@
+    expect(container.find('.test--article__container').length).toBe(1)
+    expect(container.find('.test--article__title').length).toBe(1)
+    expect(container.find('.test--article__btn').length).toBe(1)
+    expect(container.find('test--article__body').length).toBe(0)
```
Не проверяй в одном тесте сразу все

gavrilenco replied:
fixed

romabelka commented on src/components/comment-list.spec.js:
```diff
@@ -0,0 +1,43 @@
+import articles from '../fixtures'
+import CommentList from './comment-list-hooks'
+
+Enzyme.configure({ adapter: new Adapter() })
```
Можно засетапить один раз в `src/setupTests.js`

gavrilenco replied:
fixed

romabelka commented on src/components/forms/comment-add.js:
```diff
@@ -0,0 +1,67 @@
+
+  const [comment, setComment] = useState('')
+  const [username, setUsername] = useState('')
+  const [userValidationResult, setUserValidation] = useState(defaultValidationResult)
```
Лучше хранить минимальное состояние, а есть ошибка, или нет - ты можешь всегда при неодходимости посчитать

gavrilenco replied:
fixed

Commit:
HOMEWORK_2

	- added tests setup to one file
	- changed validation state to be computable
	- broke tests into smaller ones
